### PR TITLE
 planner/core: speed up unit-test

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Speed up unit test.

### What is changed and how it works?

Reduce sleep time.
Load stats instead of inserting and analyzing.

### Check List 

Tests 

 - Unit test

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8598)

### CODE REVIEW

File: planner/core/prepare_test.go [hunk 1]
```diff
@@ -205,25 +205,15 @@ func (s *testPrepareSuite) TestPrepareCacheNow(c *C) {
...
-
-	tk.MustExec(`prepare stmt4 from "select unix_timestamp(), sleep(1), unix_timestamp()"`)
-	// When executing one statement at the first time, we don't use cache, so we need to execute it at least twice to test the cache.
-	rs = tk.MustQuery("execute stmt4").Rows()
-	c.Assert(rs[0][0].(string), Equals, rs[0][2].(string))
+	c.Assert(rs[0][0].(string), Equals, rs[0][5].(string))
+	c.Assert(rs[0][1].(string), Equals, rs[0][6].(string))
+	c.Assert(rs[0][2].(string), Equals, rs[0][7].(string))
+	c.Assert(rs[0][3].(string), Equals, rs[0][8].(string))
+	c.Assert(false, IsTrue)
```
Comment (XuHuaiyu) [inline planner/core/prepare_test.go]: what does this mean?

Comment (winoros) [reply planner/core/prepare_test.go]: For testing... removed

File: planner/core/cbo_test.go [hunk 2]
```diff
@@ -246,14 +268,12 @@ func (s *testAnalyzeSuite) TestIndexRead(c *C) {
 	testKit.MustExec("create index e on t (e)")
 	testKit.MustExec("create index b_c on t (b,c)")
 	testKit.MustExec("create index ts on t (ts)")
-	for i := 0; i < 100; i++ {
-		testKit.MustExec(constructInsertSQL(i, 100))
-	}
 	testKit.MustExec("create table t1 (a int, b int, index idx(a), index idxx(b))")
+	err = s.loadTableStats("analyzesSuiteTestIndexReadT.json", dom)
```
Comment (lamxTyler) [inline planner/core/cbo_test.go]: Where is the file?

Comment (XuHuaiyu) [inline planner/core/cbo_test.go]: where is this json file?

**Comment by zz-jason** [comment]
@winoros How much time does this PR improved?

File: planner/core/cbo_test.go (hunk 2 above)
Comment (winoros) [reply planner/core/cbo_test.go]: added

**Comment by winoros** [comment]
12s-4s

File: planner/core/cbo_test.go (hunk 2 above)
Comment (XuHuaiyu) [reply planner/core/cbo_test.go]: Maybe we need to define this filename as a constant.
And add detailed comments for the statistics in it.

Comment (winoros) [reply planner/core/cbo_test.go]: Emmm why choose constant?

File: planner/core/cbo_test.go [hunk 3]
```diff
@@ -246,14 +268,12 @@ func (s *testAnalyzeSuite) TestIndexRead(c *C) {
...
+	c.Assert(err, IsNil)
 	for i := 1; i < 16; i++ {
```
[the remark this answers is not in the document]
Comment (winoros) [reply planner/core/cbo_test.go]: Yes we can, but insert little data doesn't affect too much. 
Actually in this test file there're many other places inserting and analyzing a small dataset.

**Comment by winoros** [comment]
Emmm... CicleCI's `now(3)` result limited to second... Seems that you cannot get the part of millisecond...
So the sleep duration is still 1 second in that case...

**Comment by winoros** [comment]
Emmm... CicleCI's `now(3)` result limited to second... Seems that you cannot get the part of millisecond...
So the sleep duration is still 1 second in that case...

File: planner/core/cbo_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (zz-jason) [reply planner/core/cbo_test.go]: OK

**Comment by zz-jason** [comment]
@XuHuaiyu @eurekaka PTAL

File: planner/core/cbo_test.go (hunk 2 above)
Comment (eurekaka) [reply planner/core/cbo_test.go]: I guess he means that we need to add some comments to illustrate how the statistics was generated.

**Comment by XuHuaiyu** [comment]
@winoros It's not expected.
It seems that we should not build retType like that. Since we would lost `Flen` and `Decimal` and cause unexpected error.

**Comment by winoros** [comment]
created one issue https://github.com/pingcap/tidb/issues/8615.
Currently, just sleep 1s.

**Comment by winoros** [comment]
waiting for #8619

**Comment by winoros** [comment]
After that pr merged, now it's ok to merge this one.
PTAL @XuHuaiyu @lamxTyler @eurekaka

**Pull request merged**